File: csharp/Adapter/AdapterPattern.cs
namespace design_pattern.csharp.Adapter
{
    internal class Sparrow : IBird
    {
        public void Fly() => Console.WriteLine("Flying...");

        public void MakeSound() => Console.WriteLine("Chirp..Chirp...");
    }

    internal class PlayDuck : IToyDuck
    {
        public void Squeak() => Console.WriteLine("Squeak...");
    }

    internal class BirdAdapter : IToyDuck
    {
        public IBird Bird { get; }

        public BirdAdapter(IBird bird)
        {
            Bird = bird;
        }

        public void Squeak() => Bird.MakeSound();
    }
}

File: python/adapter_pattern.py


class Bird:
    def fly(self):
        print('Bird Flying...')
    
    def make_sound(self):
        print('Bird chirp...chirp..')

class ToyDuck:
    def squeak(self):
        print('Toy Squeaking...')

class BirdAdapter(object):
    def __init__(self, object):
        self.object = object
    
    def squeak(self):
        print('Adapter.....')
        self.object.make_sound()

bird = Bird()
toy = ToyDuck()

# adapter
birdAdapter = BirdAdapter(bird)

print(bird.fly())
print(bird.make_sound())

print(toy.squeak())

print(birdAdapter.squeak())